controllers/storageclassclaim/storageclassclaim_controller.go

```diff
@@ -199,6 +200,18 @@ func (r *StorageClassClaimReconciler) reconcileConsumerPhases() (reconcile.Resul
 			},
 		}
 		if err = r.get(existing); err == nil {
+			if annotation, found := existing.GetAnnotations()[StorageClassClaimAnnotation]; found {
+				claimNamespacedName := strings.Split(annotation, "/")
+				if claimNamespacedName[0] != r.storageClassClaim.Namespace {
+					r.log.Error(fmt.Errorf("storageClass belongs to %q storageclass claim in %q namespace", r.storageClassClaim.Name, r.storageClassClaim.Namespace), "StorageClassClaim validation failed.")
+					r.storageClassClaim.Status.Phase = v1alpha1.StorageClassClaimFailed
+					err = r.update(r.storageClassClaim)
+					if err != nil {
+						r.log.Error(err, "failed to update StorageClassClaim", "storageClassClaim", klog.KRef(r.storageClassClaim.Namespace, r.storageClassClaim.Name))
+					}
+					return reconcile.Result{}, err
+				}
+			}
```

Comment:
Shouldn't we also address the case where the annotation does not exist?

Reply:
what we should do in this case?

Reply:
I would say we should fail the validation